hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/sklearn/LocalhostSkLearnServingController.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.common.serving.sklearn;
+
+import com.google.common.base.Strings;
+import com.google.common.io.Files;
+import io.hops.hopsworks.common.dao.hdfs.inode.InodeFacade;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.serving.Serving;
+import io.hops.hopsworks.common.dao.serving.ServingFacade;
+import io.hops.hopsworks.common.dao.user.Users;
+import io.hops.hopsworks.common.hdfs.Utils;
+import io.hops.hopsworks.common.security.CertificateMaterializer;
+import io.hops.hopsworks.common.serving.ServingException;
+import io.hops.hopsworks.common.serving.ServingWrapper;
+import io.hops.hopsworks.common.util.OSProcessExecutor;
+import io.hops.hopsworks.common.util.ProcessDescriptor;
+import io.hops.hopsworks.common.util.ProcessResult;
+import io.hops.hopsworks.common.util.Settings;
+import io.hops.hopsworks.restutils.RESTCodes;
+
+import javax.ejb.EJB;
+import javax.ejb.Stateless;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.enterprise.inject.Alternative;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static io.hops.hopsworks.common.hdfs.HdfsUsersController.USER_NAME_DELIMITER;
+import static io.hops.hopsworks.common.serving.LocalhostServingController.PID_STOPPED;
+import static io.hops.hopsworks.common.serving.LocalhostServingController.SERVING_DIRS;
+
+/**
+ * Localhost SkLearn Serving Controller
+ *
+ * Launches/Kills a local Flask Server for Serving SkLearn Models
+ */
+@Alternative
```

Comment:
Same as for the `LocalhostTfServingController` - This class does not implement any interface, so it should not be marked as `alternative`

Reply:
Done